core/base/src/TObject.cxx

```diff
@@ -23,6 +23,16 @@ ROOT collection classes.
 TObject's bits can be used as flags, bits 0 - 13 and 24-31 are
 reserved as  global bits while bits 14 - 23 can be used in different
 class hierarchies (watch out for overlaps).
+
+Note:
+   Class inheriting directly or indirectly from TObject should not use
+   `= default` for any of the constructors.
+   Default implementation can do 'more' than expected.  In this case it reset all the
+   data member of `TNamed` and more importantly its base class `TObject`.
```

Comment:
Why are you talking about `TNamed` specifically here, when this affects any derived class? Please explain that more clearly.

Reply:
It was a left over from the doc I had proposed for TNamed :).  I have updated the text.